web-local/src/lib/svelte-query/invalidation.ts

```diff
@@ -100,20 +100,30 @@ export const invalidateMetricsViewData = (
   queryClient: QueryClient,
   metricsViewName: string
 ) => {
+  // remove inactive queries, this is needed since these would be re-fetched with incorrect filter
+  queryClient.removeQueries({
```

Comment:
Since those old filter queries are already invalid it makes sense to just remove them. The reason we cannot refetch them is the filter object used in making the query has updated now and we would get incorrect data for the wrong API.

Reply:
Without `removeQueries`, I see two scenarios:
1. If a QueryObserver is added, then the queries will turn active, and, given invalidateQueries below, the queries will be refetched.
2. If a QueryObserver isn't added, then the queries will be garbage collected and removed automatically.

Reply:
So, without `removeQueries`, I still don't see a scenario where we fetch incorrect data. If you still see one, mind giving a more specific example?

Reply:
I am not sure i get exactly what needs to be changed here. Do you mean we can just have `invalidateQueries` ? That doesnt work, it didnt invalidate metrics view queries.

Steps to reproduce,
1. Go to dashboard, select a few filters for a particular dimension.
2. Go to model, change filter to exclude certain values in the dimension from step 1.
3. Go back to dashboard, all values are still from before the model changes.